services/web/client/source/resource/qxapp/node-meta-v1.0.0.json

```diff
@@ -0,0 +1,331 @@
+                  "additionalProperties": false,
+                  "required": [
+                      "displayOrder",
+                      "label",
+                      "description",
+                      "type"
+                  ],
+                  "properties": {
+                      "displayOrder": {
+                          "type": "number",
+                          "description": "use this to numerically sort the properties for display",
+                          "examples": [
+                              1,
+                              -0.2
+                          ]
+                      },
+                      "label": {
+                          "type": "string",
+                          "description": "short name for the property",
+                          "examples": [
+                              "Age"
+                          ]
+                      },
+                      "description": {
+                          "type": "string",
+                          "description": "description of the property",
+                          "examples": [
+                              "Age in seconds since 1970"
+                          ]
+                      },
+                      "type": {
+                          "type": "string",
+                          "pattern": "^(number|integer|boolean|string|data:([^/\\s,]+/[^/\\s,]+|\\[[^/\\s,]+/[^/\\s,]+(,[^/\\s]+/[^/,\\s]+)*\\]))$",
+                          "description": "data type expected on this input glob matching for data type is allowed",
+                          "examples": [
+                              "number",
+                              "boolean",
+                              "data:*/*",
+                              "data:text/*",
+                              "data:[image/jpeg,image/png]",
+                              "data:application/json",
+                              "data:application/json;schema=https://my-schema/not/really/schema.json",
+                              "data:application/vnd.ms-excel",
+                              "data:text/plain",
+                              "data:application/hdf5",
+                              "data:application/edu.ucdavis@ceclancy.xyz"
+                          ]
+                      },
+                      "fileToKeyMap": {
+                          "description": "Place the data associated with the named keys in files",
+
+                          "type": "object",
+                          "patternProperties": {
+                              ".+": {
+                                  "type": "string",
+                                  "pattern": "^[_a-z0-9]+$"
+                              }
+                          },
+                          "examples": [
+                              {
+                                  "dir/input1.txt": "key_1",
+                                  "dir33/input2.txt": "key2"
```

Comment:
the idea of this element is that you could hide the filenames required by the process inside the service. both coming in and going out of the service ... this is from a concept discussion I had with @mguidon back in the day when I created the first version of this ... maybe it needs some tweaking now or is even unnecessary due to the new API access.

Reply:
well I also discussed it with him and it might still be necessarry when a service expect a specific name for a file. f.e the CC use case expects a specific file name.
What is unclear is why this is a map as it is contained in the port? Why not a list?

Reply:
hmmm true ... thinking of it, I guess this should not be part of the port but one level up. or else be a list ...

Reply:
ok agreed